packages/airnode-node/src/config/index.ts

```diff
@@ -23,23 +16,31 @@ function parseOises(oises: OIS[]): OIS[] {
   });
 }
 
-export function parseConfig(configPath: string, secrets: Record<string, string | undefined>): Config {
-  const config = fs.readFileSync(configPath, 'utf8');
-
-  const validationResult = validateConfig(JSON.parse(config), secrets);
-  if (!validationResult.valid) {
-    throw new Error(`Invalid Airnode configuration file: ${JSON.stringify(validationResult.messages)}`);
+const readConfig = (configPath: string): unknown => {
+  try {
+    return JSON.parse(fs.readFileSync(configPath, 'utf8'));
+  } catch (e) {
+    throw new Error('Failed to parse config file');
   }
+};
 
-  const interpolatedConfig = template(config, {
-    escape: NO_MATCH_REGEXP,
-    evaluate: NO_MATCH_REGEXP,
-    interpolate: ES_MATCH_REGEXP,
-  })(secrets);
-  const parsedConfig = JSON.parse(interpolatedConfig);
+export interface ParseConfigResult {
+  config: Config;
+  validationOutput: Result;
+  shouldSkipValidation: boolean;
+}
 
+export function parseConfig(configPath: string, secrets: Record<string, string | undefined>): ParseConfigResult {
+  const config = readConfig(configPath);
+  const validationResult = validateConfig(config, secrets);
+  const parsedConfig: Config = validationResult.specs as Config;
   const ois = parseOises(parsedConfig.ois);
-  return { ...parsedConfig, ois };
+
+  return {
+    config: { ...parsedConfig, ois },
+    shouldSkipValidation: !!parsedConfig.nodeSettings.skipValidation,
+    validationOutput: validationResult,
+  };
 }
```

Comment:
I know that this way we can do whatever with the results (throw, use different loggers, etc.) but I wouldn't be against doing that within this function. But I guess that would be a bigger refactor. Just a thought...

Reply:
The problem is that deployer wants to log it and node would like to do that as well, but for now doesn't.

Fundamentally, they use a different type of loggers (and they probably always will) so it's hard to do and actually I'd say out of scope of this function.